### ISSUE TITLE
Dialog header customization

### ISSUE THREAD

**Issue by origami-z** [jpmorganchase/salt-ds#3357]
### Area

UI Components

### The problem

Dialog header text and icon size are fixed to H2 and size=1, in some application, text and icon needs to be bigger.

~~We also want to make Dialog footer "stick" to the bottom, when trying to resize the dialog.~~ Replaced in #3435

### The solution

Design discussion needed

### Alternatives and examples

/

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.

**Comment by JoanaMMoreira** [comment]
Closing this and creating a new issue as we no longer need to change the header but will need to fix the footer alignment

**Comment by origami-z** [comment]
Reopen, still marked as blocker by our user

**Comment by mark-tate** [comment]
Goal: 
Identify what we need to do to unblock MTK

**Comment by origami-z** [comment]
It's possible to achieve a different header text size
```
        <DialogHeader
          header={<Text styleAs="display1">Subscribe</Text>}
          preheader="Recieve emails about the latest updates"
        />
```


https://codesandbox.io/p/sandbox/goofy-voice-fvh4xp?file=%2Fsrc%2FPreheader.jsx

**Comment by origami-z** [comment]
Closing given the functionality can be done in [above comment](https://github.com/jpmorganchase/salt-ds/issues/3357#issuecomment-2195081503)
